### DISCUSSION TITLE
math mode

### DISCUSSION THREAD

**Discussion by esquires** [nvim-neorg/neorg#438] [Ideas]
Hi all, thanks again for a great plugin! It is fantastic.

After using neorg a bit I was wondering if the following would be easy to add:

* concealing for math characters (e.g., \lambda would render as the greek character, x_1 would render with a subscript, etc): A list of examples of conceal types for math mode from vimtex is [here](https://github.com/lervag/vimtex/blob/master/doc/vimtex.txt#L2199)
* labelling equations in `@math` blocks so we could do something like this:

```
@math 
x = y + 1
\label{eq:my-eqn}
@end

@math 
z = \lambda^2
\label{eq:my-second-eqn}
@end

My equation is (\ref{eq:my-eqn}) and (\ref{eq:my-second-eqn}).
```

It would be super neat if the `@math` block could render the math blocks as like `x = y + 1    (1)`, `z = \lambda^2    (2)` and then the following line as `My equation is (1) and (2)`. It would also be great to add the `\ref` to the autocomplete so we can select from a list of previously defined equations.

Not sure if this is possible or difficult to do. Prior [discussion](https://github.com/nvim-neorg/neorg/issues/292) on the concealer sound like it has been pretty involved (thanks for work on that as well). 

Happy to help if you would like.

**Comment by mrossinek** [comment]
This is definitely planned for the future, so thanks for raising this topic for discussion!
We already had some chats revolving around an integration with [nabla.nvim](https://github.com/jbyuki/nabla.nvim) but no serious efforts have been made yet. We really want to get the 0.1 release out the door, but this is high on our priority list for 0.2 :+1:
